pfor doesn't catch panics

@rsc The parallel execution introduced by #84 doesn't seem to guard against the possibility of legitimate exceptions such as division by zero.

For example:

```
% ivy
1 / 200 6 rho 0
panic: division by zero

goroutine 46 [running]:
robpike.io/ivy/value.Errorf(...)
	/home/rogpeppe/src/go/src/github.com/rsc/ivy/value/value.go:49
robpike.io/ivy/value.init.0.func13({0x57cd88?, 0x543509?}, {0x57cce8?, 0xc000126400?}, {0x57cce8?, 0xc000126440?})
	/home/rogpeppe/src/go/src/github.com/rsc/ivy/value/binary.go:277 +0xab
robpike.io/ivy/value.(*binaryOp).EvalBinary(0xc0001242a0, {0x57cab8, 0xc000125260}, {0x57cd88, 0x632148}, {0x57cd88, 0x632140})
	/home/rogpeppe/src/go/src/github.com/rsc/ivy/value/eval.go:112 +0x2a8
robpike.io/ivy/exec.(*Context).EvalBinary(0xc000125260, {0x57cd88, 0x632148}, {0x543509, 0x1}, {0x57cd88, 0x632140})
	/home/rogpeppe/src/go/src/github.com/rsc/ivy/exec/context.go:172 +0x189
robpike.io/ivy/value.binaryMatrixOp.func1(0x0?, 0x4b0)
	/home/rogpeppe/src/go/src/github.com/rsc/ivy/value/eval.go:597 +0xfb
robpike.io/ivy/value.pfor.func1()
	/home/rogpeppe/src/go/src/github.com/rsc/ivy/value/eval.go:188 +0x67
created by robpike.io/ivy/value.pfor
	/home/rogpeppe/src/go/src/github.com/rsc/ivy/value/eval.go:186 +0x107
```

--- a/testdata/exec_fail.ivy
+++ b/testdata/exec_fail.ivy
@@ -47,3 +47,9 @@ x
 # index must be integer
 1 2[1 2 log 2]
 	X
+
+1 / 0
+	X
+
+1 / 2 2 rho 0
+	X

--- a/value/eval.go
+++ b/value/eval.go
@@ -7,7 +7,6 @@ package value
 import (
 	"runtime"
 	"strings"
-	"sync"
 )
 
 type valueType int
@@ -179,16 +178,27 @@ func pfor(ok bool, size, n int, f func(lo, hi int)) {
 	if q := n * size / pforMinWork; q < p {
 		p = q
 	}
-	var wg sync.WaitGroup
+	c := make(chan interface{}, p)
 	for i := 0; i < p; i++ {
 		lo, hi := i*n/p, (i+1)*n/p
-		wg.Add(1)
 		go func() {
-			defer wg.Done()
+			defer sendRecover(c)
 			f(lo, hi)
 		}()
 	}
-	wg.Wait()
+	var bad interface{}
+	for i := 0; i < p; i++ {
+		if e := <-c; e != nil {
+			bad = e
+		}
+	}
+	if bad != nil {
+		panic(bad)
+	}
+}
+
+func sendRecover(c chan<- interface{}) {
+	c <- recover()
 }
 
 // inner product computes an inner product such as "+.*".

